lib/src/lib.rs

```diff
@@ -297,6 +297,35 @@ pub fn decrypt(
     Ok(djb::decrypt(key, data, ctx.as_bytes())?)
 }
 
+pub fn prove_blake3_work(nonce: &[u8], cost: u8) -> Option<u32>
+{
+    let maxn: u32 = u32::MAX - 1;
+    let mut n: u32 = 0;
+    while n < maxn {
```

Comment:
I don't see why this can't be u32::MAX. But ideally we choose a much lower limit because it would take a rather long time to do 4billion iterations of this.

Reply:
I'm inclined to adjust this when we tune the cost.